### PR TITLE
Merge patient lab iips mocha

### PR DESCRIPTION
**Pull request by catch-point**
Update merge-patient-lab-iips-mocha to not use rpf object

### CODE REVIEW

File: src/noflo-component-access.js [hunk 1]
```diff
@@ -34,75 +36,14 @@ function facadePort(nodeInstance, port, name) {
...
-    if (idx >= 0) return fn._bind_facade_noflo_access[idx + 1];
-    fn._bind_facade_noflo_access.push(this, fn.bind(this));
-    return _.last(fn._bind_facade_noflo_access);
+        isMulti: port.isAddressable.bind(port),
+        isRequired: port.isRequired.bind(port),
+        listAttached: port.listAttached.bind(port)
+    }, isOutPort(port) ? {
+        connect: port.connect.bind(port),
+        send: port.send.bind(port),
+        disconnect: port.disconnect.bind(port)
```
Comment (dbooth-boston) [inline src/noflo-component-access.js:47]: I would rather not expose connect, send and disconnect to the RDF Pipeline layer.  Can you expose only a sendState function instead, which does the noflo connect-send-disconnect sequence?  Or do you temporarily need access to the individual connect, send and disconnect functions for some of the existing test code?

File: test/common-test.js [hunk 2]
```diff
@@ -57,10 +68,11 @@ module.exports = {
             port.name = name;
         });
 
-        return component;
+        return node ? node : component;
     },
 
-    sendData: function(component, port, payload) {
+    sendData: function(node, port, payload) {
```
Comment (dbooth-boston) [inline test/common-test.js:83]: I think this should be in the access layer instead.

**Comment by dbooth-boston** [comment]
LGTM aside from my comments about moving sendData to the access layer.

File: test/merge-patient-lab-iips-mocha.js [hunk 3]
```diff
@@ -8,181 +9,146 @@ chai.use(chaiAsPromised);
...
+                    data: {
+                        id: '001',
+                        name: 'Alice',
+                        dob: '1979-01-23',
+                        glucose: '75',
+                        date: '2012-02-01'
+                    },
+                    lm: 'LM1328113669.00000000000000001'
+                };
+                node.vni().outputState( _.mapObject(outputState, _.clone) );
```
Comment (dbooth-boston) [inline test/merge-patient-lab-iips-mocha.js:146]: JS noobie question: could this line just as well have been written ass

<pre>
node.vni().outputState( _.clone(outputState) );
</pre>

File: test/merge-patient-lab-iips-mocha.js [hunk 4]
```diff
@@ -8,181 +9,146 @@ chai.use(chaiAsPromised);
...
+
+                var currentState = node.vni().outputState();
+                currentState.should.be.an('object');
+                currentState.data.should.deep.eql( outputState.data );
```
Comment (dbooth-boston) [inline test/merge-patient-lab-iips-mocha.js:150]: Why isn't the whole state being compared?

File: test/noflo-component-factory-mocha.js [hunk 5]
```diff
@@ -145,7 +118,55 @@ describe('noflo-component-factory', function() {
                     }
                 }
             }));
-            test.sendData(component, 'input', "zero");
+            test.sendData(node, 'input', "zero");
         }).should.become(0);
     });
+    it("should has a nodeName", function() {
```
Comment (dbooth-boston) [inline test/noflo-component-factory-mocha.js]: s/has/have/

**Comment by dbooth-boston** [comment]
LGTM subject to minor comments

File: test/merge-patient-lab-iips-mocha.js (hunk 3 above)
Comment (catch-point) [inline test/merge-patient-lab-iips-mocha.js:146]: _.clone is a shallow copy. By using _.mapObject with _.clone the copy goes an extra level deep.

Comment (dbooth-boston) [inline test/merge-patient-lab-iips-mocha.js:146]: thanks

File: test/merge-patient-lab-iips-mocha.js (hunk 4 above)
Comment (catch-point) [inline test/merge-patient-lab-iips-mocha.js:150]: I don't know, Glenna?

Comment (dbooth-boston) [inline test/merge-patient-lab-iips-mocha.js:150]: Could it compare the whole state, like this?

<pre>
currentState.should.deep.eql( outputState );
</pre>

Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js:150]: Yes, you could.

Comment (catch-point) [inline test/merge-patient-lab-iips-mocha.js:150]: I think that might enforce some property order that isn't really required.

**Comment by dbooth-boston** [comment]
LGTM

File: test/merge-patient-lab-iips-mocha.js [hunk 6]
```diff
@@ -8,181 +9,146 @@ chai.use(chaiAsPromised);
 var _ = require('underscore');
 var noflo = require('noflo');
 
-var componentFactory = require('../components/merge-patient-lab-iips');
-var getComponent = componentFactory.getComponent;
+var compFactory = require('../components/merge-patient-lab-iips');
 
-var commonTest = require('./common-test');
+var test = require('./common-test');
 var commonStubs = require('./common-stubs');
```
Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js]: If you rename the commonTest to test, perhaps commonStubs should be simply stubs.

File: test/merge-patient-lab-iips-mocha.js [hunk 7]
```diff
@@ -8,181 +9,146 @@ chai.use(chaiAsPromised);
...
  
-        it("should have an rpf object on nodeInstance", function() {
-            return Promise.resolve(componentFactory.getComponent )
-                .then(commonTest.createComponent).then(function(component){
-                    component.rpf.should.exist;
-                    component.rpf.should.be.an('object');
-            });
-        });
+    it("should be an object on nodeInstance", function() {
+        var node = test.createComponent(compFactory);
```
Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js:20]: Looks like test.createComponent(compFactory) is done in every test.   That seems like a candidate for moving into a beforeEach function.

File: test/merge-patient-lab-iips-mocha.js [hunk 8]
```diff
@@ -8,181 +9,146 @@ chai.use(chaiAsPromised);
...
+                var patientState = {
+                    data: { id: '001', name: 'Alice', dob: '1979-01-23' },
+                    lm: 'LM1328113669.00000000000000001'
+                };
+                node.vni().inputStates({'patient':
+                                                  _.mapObject(patientState, _.clone)});
+
+                var currentState = node.vni().inputStates('patient');
+                currentState.should.be.an('object');
+		        currentState.data.should.eql( patientState.data );
```
Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js]: Odd indentation here

File: test/merge-patient-lab-iips-mocha.js (hunk 4 above)
Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js:150]: True.  That's one of the reasons I usually test individual state elements one by one (plus then it's easier to see which one failed).

File: test/merge-patient-lab-iips-mocha.js (hunk 7 above)
Comment (catch-point) [inline test/merge-patient-lab-iips-mocha.js:20]: yes, but since it is only one line of code, I'd rather keep it as a test-local variable.

Comment (glenna-m) [inline test/merge-patient-lab-iips-mocha.js:20]: Since it's one line, that's fine.

**Comment by glenna-m** [comment]
LGTM

**Pull request merged**